src/promql/src/functions/deriv.rs

```diff
@@ -12,5 +12,121 @@
 // See the License for the specific language governing permissions and
 // limitations under the License.
 
-//! Implementation of [`reset`](https://prometheus.io/docs/prometheus/latest/querying/functions/#deriv) in PromQL. Refer to the [original
+//! Implementation of [`deriv`](https://prometheus.io/docs/prometheus/latest/querying/functions/#deriv) in PromQL. Refer to the [original
 //! implementation](https://github.com/prometheus/prometheus/blob/90b2f7a540b8a70d8d81372e6692dcbb67ccbaaa/promql/functions.go#L839-L856).
+
+use std::sync::Arc;
+
+use common_function_macro::range_fn;
+use datafusion::arrow::array::{Float64Array, TimestampMillisecondArray};
+use datafusion::arrow::datatypes::TimeUnit;
+use datafusion::common::DataFusionError;
+use datafusion::logical_expr::{ScalarUDF, Signature, TypeSignature, Volatility};
+use datafusion::physical_plan::ColumnarValue;
+use datatypes::arrow::array::Array;
+use datatypes::arrow::datatypes::DataType;
+
+use crate::functions::{compensated_sum_inc, extract_array};
+use crate::range_array::RangeArray;
+
+#[range_fn(name = "Deriv", ret = "Float64Array", display_name = "prom_drive")]
+pub fn drive(times: &TimestampMillisecondArray, values: &Float64Array) -> Option<f64> {
+    if values.len() < 2 {
+        None
+    } else {
+        let intercept_time = times.value(0);
+        let (slope, _) = linear_regression(times, values, intercept_time);
+        slope
+    }
+}
+
+fn linear_regression(
```

Comment:
This is relatively complex, would you mind adding some documentation about it? Like what it calculates or where you refer from

Reply:
ok